call-automation-appointment-reminder/program.ts

```diff
@@ -0,0 +1,350 @@
+import { CommunicationIdentityClient } from "@azure/communication-identity";
+import {
+  CommunicationIdentifier,
+  CommunicationUserIdentifier,
+  PhoneNumberIdentifier,
+} from "@azure/communication-common";
+import { MessageType, Logger } from "./Logger";
+import { CallConfiguration } from "./CallConfiguration";
+import {
+  CallAutomationClient,
+  CreateCallOptions,
+  CreateCallResult,
+  CallAutomationEventParser,
+  CallAutomationEvent,
+  CallMediaRecognizeDtmfOptions,
+  DtmfTone,
+  RecognizeInputType,
+  RecognizeCompleted,
+  FileSource,
+  CallInvite,
+} from "@azure/communication-call-automation";
+import { CloudEvent } from "@azure/eventgrid";
+import { Request, Response } from "express";
+
+var configuration = require("./config");
+var express = require("express");
+var router = express.Router();
+var fileSystem = require("fs");
+var path = require("path");
+
+var url = "http://localhost:8080";
+var callConfiguration: CallConfiguration;
+var callAutomationClient: CallAutomationClient;
+var callConnection: CreateCallResult;
+var callAutomationEventParser = new CallAutomationEventParser();
+var userIdentityRegex = new RegExp(
+  "8:acs:[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}_[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}"
+);
+var phoneIdentityRegex = new RegExp("^\\+\\d{10,14}$");
+enum CommunicationIdentifierKind {
+  UserIdentity,
+  PhoneIdentity,
+  UnknownIdentity,
+}
+var appBaseUrl: string = configuration.AppBaseUri;
+callAutomationClient = new CallAutomationClient(configuration.ConnectionString);
+callConfiguration = initiateConfiguration(appBaseUrl);
+var identifierKind = getIdentifierKind(configuration.TargetPhoneNumber);
+
+async function runSample() {
+  try {
+    var sourcePhoneNumber: PhoneNumberIdentifier = {
+      phoneNumber: callConfiguration.sourcePhoneNumber,
+    };
+    var targetIdentity = targetUserIdentity();
+    var callInviteOptions: CallInvite =
+      identifierKind == CommunicationIdentifierKind.PhoneIdentity
```

Comment:
we can use **isPhoneNumberIdentifier(targetIdentity)** for comparting the type of object, it will also resolves the error we got in visual studio.